addons/sale/models/payment_transaction.py

```diff
@@ -20,11 +20,19 @@ class PaymentTransaction(models.Model):
     def _compute_sale_order_reference(self, order):
         self.ensure_one()
         if self.provider_id.so_reference_type == 'so_name':
-            return order.name
+            order_reference = order.name
         else:
             # self.provider_id.so_reference_type == 'partner'
             identification_number = order.partner_id.id
-            return '%s/%s' % ('CUST', str(identification_number % 97).rjust(2, '0'))
+            order_reference = '%s/%s' % ('CUST', str(identification_number % 97).rjust(2, '0'))
+
+        invoice_journal = self.env['account.journal'].search([('type', '=', 'sale'), ('company_id', '=', self.env.company.id)], limit=1)
+        if invoice_journal.invoice_reference_model == 'ch':
```

Comment:
Do not put custom code only affecting `ch` in the sale module. You should create a `payment_transaction.py` in the l10n_ch module and override the function.

Reply:
But l10n_ch does not depend on payment, does it ? Though, I agree with you ; that's why a suggested a hook in account instead :)

Reply:
`l10n_ch_sale` then :D

Reply:
> `l10n_ch_sale` then :D

It was the first solution. To avoid the creation/installation of a whole module just to get the correct reference for l10n_ch, we thought it was better to implement a hook (not well implemented at first due to a slight misunderstanding on my side)